src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFChartData.java

```diff
@@ -98,6 +98,10 @@ public XDDFCategoryAxis getCategoryAxis() {
 
         protected abstract CTNumDataSource getNumDS();
 
+        protected abstract void updateIdXVal(long val);
+        
+        protected abstract void updateOrderVal(long val);
```

Comment:
Soooooo....

They were initially intended to be `protected`, not `public`!

We need to explore more on these, after fixing all other details.

Reply:
We need this methdos as public so updated code.